File: JavaScript/Books/Entendendo Algoritmos/buscaBinaria.js
function pesquisaBinaria(lista, item) {
    let baixo = 0;
    let alto = lista.length - 1;

    while (baixo <= alto) {
        let meio = parseInt((baixo + alto) / 2);
        let chute = lista[meio];

        if (chute === item) {
            return meio;
        } else if (chute > item) {
            alto = meio - 1;
        } else {
            baixo = meio + 1;
        }

    }

    return "Não encontrado";

}

numeros = [1, 3, 5, 7, 9];
console.log(pesquisaBinaria(numeros, 3)) // 1
console.log(pesquisaBinaria(numeros, 2)) // Não encontrado
console.log(pesquisaBinaria(numeros, 10)) // Não encontrado
console.log(pesquisaBinaria(numeros, 7)) // 3

File: Python/Estudo/Livro - Entendendo Algoritmos/buscaBinaria.py
def pesquisaBinaria(lista, item):
    baixo = 0
    alto = len(lista) - 1

    while baixo <= alto:
        meio = int((baixo + alto) / 2)
        chute = lista[meio]

        print('Tentativa')
        if chute == item:
            return meio
        elif chute > item:
            alto = meio - 1
        else:
            baixo = meio + 1
    
    return None

numeros = [1, 3, 5, 7, 9]
print(pesquisaBinaria(numeros, 3)) # 1
print(pesquisaBinaria(numeros, 2)) # None
print(pesquisaBinaria(numeros, 1)) # 0
print(pesquisaBinaria(numeros, 9)) # 4

arrayGrande = list(range(1, 129))
print(pesquisaBinaria(arrayGrande, 1)) # 7 Tentativas

arrayGrande = list(range(1, 257))
print(pesquisaBinaria(arrayGrande, 1)) # 8 Tentativas